projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php

```diff
@@ -76,26 +94,53 @@ public function reregister_menu_items() {
 			$this->add_new_site_link();
 		}
 
+		// Add WP Admin menu on Calypso when SSO is disabled.
+		if ( $this->is_api_request && ! $this->is_sso_enabled ) {
+			$this->add_wp_admin_menu();
+		}
+
 		ksort( $GLOBALS['menu'] );
 	}
 
 	/**
-	 * Forces Posts menu to WPAdmin for Atomic sites only.
-	 * Overloads `add_posts_menu` in parent class.
+	 * Whether to use wp-admin pages rather than Calypso.
+	 *
+	 * @return bool
+	 */
+	public function should_link_to_wp_admin() {
+		// Force Calypso when SSO is disabled.
+		if ( $this->is_api_request && ! $this->is_sso_enabled ) {
```

Comment:
Nitpick: Maybe we could wrap this condition into a helper method like is_api_request_with_sso_disabled and replace the usages of the condition?

Reply:
Good idea! Done in 3f35b06.